src/software/util/typename/typename.cpp

```diff
@@ -0,0 +1,8 @@
+#include "software/util/typename/typename.h"
+
+std::string demangle_typeid(const char* mangled_name)
+{
+    std::unique_ptr<char, void (*)(void*)> demangled_name_ptr{
```

Comment:
I think using a deleter is a bit opaque and overkill if we're just getting a `char*`, copying it into a `std::string`, and then `free`ing it

Reply:
ok simplified it